gradle/lib/dependabot/gradle/distributions.rb

```diff
@@ -0,0 +1,22 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "dependabot/registry_client"
+
+module Dependabot
+  module Gradle
+    module Distributions
+      extend T::Sig
+
+      DISTRIBUTIONS_URL = "https://services.gradle.org"
```

Comment:
Can we detect this from the `gradle-wrapper.properties`? 

In my current configuration, for example, this is pointing to a custom proxy. Example

```properties
distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
distributionUrl=https://example.com/gradle-distributions/gradle-9.0.0-bin.zip
validateDistributionUrl=true
zipStoreBase=GRADLE_USER_HOME
zipStorePath=wrapper/dists
distributionBuildTime=20250731163512+0000
```

Reply:
This is pretty much intentional, because: 
- The PR is aiming to support the 95% of the cases (if not more) where people use a standard Gradle distribution
- Even if I manage to infer the base URL for a non standard distribution, there won't be any guarantee that in that same server will exist an equivalent listing endpoint https://services.gradle.org/versions/all, or it will honor it's contact 

Therefore the extra complexity to support this edge case won't pay off, at least in the scope of this PR